File: ios-native/NeedsVsWants/Design/Haptics.swift
import UIKit

/// Centralized haptic feedback. The old `ios/` port had zero haptics — this is
/// a deliberate polish addition.
///
/// - seal:   medium impact when an entry auto-seals (the "stamp" feel)
/// - warn:   notification warning on delete / 20-cap reached / wipe
/// - success: notification success on "start new sheet" / wipe complete
enum Haptics {
    static func seal() {
        UIImpactFeedbackGenerator(style: .medium).impactOccurred()
    }
    static func warn() {
        UINotificationFeedbackGenerator().notificationOccurred(.warning)
    }
    static func success() {
        UINotificationFeedbackGenerator().notificationOccurred(.success)
    }
}

File: expo/src/design/Haptics.ts
/**
 * Haptic feedback — faithful port of Haptics.swift.
 *
 * seal()   → medium impact (entry sealed)
 * warn()   → notification warning (sheet full)
 * success()→ notification success (new sheet started)
 *
 * Uses expo-haptics, which is Expo-Go-compatible.
 */
import * as Haptics from "expo-haptics";

export async function seal(): Promise<void> {
  try {
    await Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Medium);
  } catch {
    // Haptics unavailable (e.g. simulator) — no-op.
  }
}

export async function warn(): Promise<void> {
  try {
    await Haptics.notificationAsync(Haptics.NotificationFeedbackType.Warning);
  } catch {
    // no-op
  }
}

export async function success(): Promise<void> {
  try {
    await Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);
  } catch {
    // no-op
  }
}
